File: app/src/main/java/com/hiddenramblings/tagmo/amiibo/code/HexItem.java
package com.hiddenramblings.tagmo.amiibo.code;

import android.graphics.Typeface;

public class HexItem {
    String text;
    int textStyle;
    int backgroundColor;

    HexItem(String text, int textStyle, int backgroundColor) {
        this.text = text;
        this.textStyle = textStyle;
        this.backgroundColor = backgroundColor;
    }

    public HexItem(String text, int backgroundColor) {
        this(text, Typeface.BOLD, backgroundColor);
    }

    public String getText() {
        return text;
    }

    public int getTextStyle() {
        return textStyle;
    }

    public int getBackgroundColor() {
        return backgroundColor;
    }
}

File: app/src/main/java/com/hiddenramblings/tagmo/hexcode/HexItem.kt
package com.hiddenramblings.tagmo.hexcode

import android.graphics.Typeface

open class HexItem internal constructor(
    var text: String,
    var textStyle: Int,
    var backgroundColor: Int
) {

    constructor(text: String, backgroundColor: Int) : this(text, Typeface.BOLD, backgroundColor) {}
}